rustls/src/tls13/key_schedule.rs

```diff
@@ -334,6 +334,92 @@ impl KeyScheduleTraffic {
         self.ks
             .export_keying_material(&self.current_exporter_secret, out, label, context)
     }
+
+    #[cfg(feature = "extract_secrets")]
+    pub(crate) fn extract_secrets(
+        &self,
+        tx_seq: u64,
+        rx_seq: u64,
+        algo: &ring::aead::Algorithm,
+        side: crate::conn::Side,
+    ) -> Result<crate::ExtractedSecrets, Error> {
+        let expand = |secret: &hkdf::Prk, key: &mut [u8], iv: &mut [u8]| -> Result<(), Error> {
```

Comment:
It looks like this closure doesn't actually depend on any local state, so it can just be a function? It could even take two const generic parameters for the sizes of the key and IV arrays so they can be returned directly instead of being passed in?

And if that works, I'm not sure the `extract` closure still carries its weight.

Reply:
I turned expand into a function with const generics, but I still think the `extract` closure is worth it due to the client/server duplication - especially when we have to slice iv into salt + iv.